tests/processing/strategies/test_run_task.py

```diff
@@ -72,3 +95,363 @@ def test_run_task_in_threads() -> None:
+
+
+def test_parallel_run_task_terminate_workers() -> None:
+    next_step = Mock()
+
+    starting_processes = get_subprocess_count()
+    worker_processes = 2
+    manager_processes = 1
+
+    with assert_changes(
+        get_subprocess_count,
+        starting_processes,
+        starting_processes + worker_processes + manager_processes,
+    ):
+        transform_step = RunTaskWithMultiprocessing(
+            transform_payload_expand,  # doesn't matter
+            next_step,
+            num_processes=worker_processes,
+            max_batch_size=5,
+            max_batch_time=60,
+            input_block_size=4096,
+            output_block_size=4096,
+        )
+
+    with assert_changes(
+        get_subprocess_count,
+        starting_processes + worker_processes + manager_processes,
+        starting_processes,
+    ), assert_changes(lambda: int(next_step.terminate.call_count), 0, 1):
+        transform_step.terminate()
+
+
+def test_parallel_run_task_bad_messages() -> None:
+    next_step = Mock()
+
+    starting_processes = get_subprocess_count()
+    worker_processes = 5
+    manager_processes = 1
+
+    # every other message has a key
+    messages = [
+        Message(
+            Value(
+                KafkaPayload(None if i % 2 == 0 else b"key", b"\x00", []),
+                {Partition(Topic("test"), 0): Position(0, datetime.now())},
+            )
+        )
+        for i in range(9)
+    ]
+
+    # everything should be processed in parallel, 5 workers should spawn
+    with assert_changes(
+        get_subprocess_count,
+        starting_processes,
+        starting_processes + worker_processes + manager_processes,
+    ):
+        # create transform step with multiple processes
+        transform_step = RunTaskWithMultiprocessing(
+            function=fail_bad_messages,
+            next_step=next_step,
+            num_processes=worker_processes,
+            max_batch_size=9,
+            max_batch_time=60,
+            input_block_size=4096,
+            output_block_size=4096,
+        )
+
+        # submit 9 messages: 4 good ones 5 bad ones
+        for message in messages:
+            transform_step.submit(message)
+            transform_step.poll()
+
+    # wait for all processes to finish
+    with pytest.raises(InvalidMessages) as e_info:
+        transform_step.close()
+        transform_step.join()
+
+    # An exception should have been thrown with the 5 bad messages
+    assert len(e_info.value.messages) == 5
+    # Test exception pickles and decodes correctly
+    invalid_message = e_info.value.messages[0]
+    assert isinstance(invalid_message, InvalidRawMessage)
+    assert invalid_message.reason == NO_KEY
+    assert invalid_message.payload == str(messages[0].payload)
+    # The 4 good ones should not have been blocked
+    assert next_step.submit.call_count == 4
+
+
+def add_one(message: Message[int]) -> int:
+    return message.payload + 1
+
+
+def test_message_rejected() -> None:
```

Comment:
This is the only new test in this file. The rest is copied over from `test_transform` since ParallelTransform is now just an alias for RunTaskWithMultiprocessing.